sql/core/src/test/resources/sql-tests/inputs/cast.sql

```diff
@@ -116,3 +116,11 @@ select cast(interval '10' day as bigint);
 
 select cast(interval '-1000' month as tinyint);
 select cast(interval '1000000' second as smallint);
+
+-- cast ANSI intervals to decimals
+select cast(interval '-1' year as decimal(10, 0));
+select cast(interval '1.000001' second as decimal(10, 6));
+select cast(interval '08:11:10.001' hour to second as decimal(10, 4));
+select cast(interval '1 01:02:03.1' day to second as decimal(8, 1));
+
+select cast(interval '10.123' second as decimal(1, 0));
```

Comment:
how about `decimal(4, 2)`? will we fail or truncate?

Reply:
Don't see much different between your example and mine. In both cases, there are not enough digits to represent the value. I mean there is a problem with precision first of all, not w/ scale. But let me check.

Reply:
I added a check. The result is truncated. Please, take a look.